### PR TITLE
MM-JOURNEY-GET-ENDPOINT: Creates GET endpoint for user journeys.

### PR DESCRIPTION
**Pull request by Bottle7**
Ticket: mm-journey-get-endpoint

# Summary
- Adds GET endpoint for journeys
- Return query index 0 for get by user id

[](https://hamzamohdzubair.github.io/redant/)

- None listed.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] Dependency updates (may include refactoring of a dependencies functionality).
- [x] New feature (non-breaking change which adds functionality).

## Checklist:

- [x] I have read the checklist.

### CODE REVIEW

File: matchmaking/backend/controllers/journey.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { journeyQueries } = require('../queries');
+
+/**
+ * Get journey by user id.
+ * @author Brandon Bouchard <brandon.bouchard@gov.bc.ca | brandonjbouchard@gmail.com>
```
Comment (adamkroonbcgov) [inline matchmaking/backend/controllers/journey.js]: nit: gov email only imo...?

File: matchmaking/backend/controllers/journey.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * @method GET
+ * @param {string} id - User GUID, a unique identifier for a user.
+ */
+exports.get_journey = async (req, res) => {
+  try {
+    const journey = await journeyQueries.getJourneyByUserId(req.params.id);
+
+    if (journey) res.status(200).json(journey); // Success, return journey.
```
Comment (adamkroonbcgov) [inline matchmaking/backend/controllers/journey.js]: were you doing pseudo here? I'd love to talk about how we structure commenting later as a group

File: matchmaking/backend/queries/journey.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const db = require('../database/connection');
+
+/**
+ * Get user by id.
+ * @author Brandon Bouchard <brandon.bouchard@gov.bc.ca | brandonjbouchard@gmail.com>
```
Comment (adamkroonbcgov) [inline matchmaking/backend/queries/journey.js]: same nit here

**Review by BradyMitch — APPROVED** [review]
Looks great!

**Comment by adamkroonbcgov** [comment]
2 smol things, but looking good.

File: matchmaking/backend/controllers/journey.js (hunk 2 above)
Comment (BradyMitch) [reply matchmaking/backend/controllers/journey.js]: Whats wrong w the comments?

**Pull request merged**